build-scripts/travis

```
@@ -209,6 +209,23 @@ case "$JOB" in
         sudo prove -v *.test
     ;;
 
+    ( schema-test )
+        # NOTE: these tests will manipulate the system "cfdb" database
+        # so parallel execution is not possible
+        s3cmd get --no-progress $DEB_PACKAGE cfe.deb
+        sudo dpkg -i cfe.deb || (
+            echo "Problem installing CFEngine package $?"
+            # TODO CFEngineHub-Install.log or CFEngine-Install.log?
+            echo "=== BEGIN /var/log/CFEngine*-Install.log ==="
+            sudo cat /var/log/CFEngine*-Install.log
+            echo "=== END /var/log/CFEngine*-Install.log ==="
+            ) # TODO bail out early?
```

Yes, why should it continue? AFAICT, it should just fail.

---

I agree. if CFEngine installs, but `dpkg` returs non-0 status - it's better to be aware of this. And green Travis build is the best way to become **un**aware :-)

---

Agreed. I think I had put an exit in there and now realize it wasn't working because I was exiting the sub-shell. Will fix! :)

---

good point. Maybe worth wrapping it into "proper" `if … then … fi` instead of "simple" `|| ( … )`?

---

I refactored to a shared `install_package` function and am testing this out. No reason to let the other JOB types fail silently.